flink-libraries/flink-table-common/src/main/java/org/apache/flink/table/functions/TableAggregateFunction.java

```diff
@@ -0,0 +1,129 @@
+ * <p>All these methods must be declared publicly, not static, and named exactly as the names
+ * mentioned above. The method {@link #createAccumulator()} is defined in
+ * the {@link UserDefinedAggregateFunction} functions, while other methods are explained below.
+ *
+ * <pre>
+ * {@code
+ * Processes the input values and update the provided accumulator instance. The method
+ * accumulate can be overloaded with different custom types and arguments. A TableAggregateFunction
+ * requires at least one accumulate() method.
+ *
+ * param: accumulator           the accumulator which contains the current aggregated results
+ * param: [user defined inputs] the input value (usually obtained from a new arrived data).
+ *
+ * public void accumulate(ACC accumulator, [user defined inputs])
+ * }
+ * </pre>
+ *
+ * <pre>
+ * {@code
+ * Retracts the input values from the accumulator instance. The current design assumes the
+ * inputs are the values that have been previously accumulated. The method retract can be
+ * overloaded with different custom types and arguments.
+ *
+ * param: accumulator           the accumulator which contains the current aggregated results
+ * param: [user defined inputs] the input value (usually obtained from a new arrived data).
+ *
+ * public void retract(ACC accumulator, [user defined inputs])
+ * }
+ * </pre>
+ *
+ * <pre>
+ * {@code
+ * Merges a group of accumulator instances into one accumulator instance. This function must be
+ * implemented for data stream session window grouping aggregates and data set grouping aggregates.
+ *
+ * param: accumulator the accumulator which will keep the merged aggregate results. It should
+ *                    be noted that the accumulator may contain the previous aggregated
+ *                    results. Therefore user should not replace or clean this instance in the
+ *                    custom merge method.
+ * param: its         an java.lang.Iterable pointed to a group of accumulators that will be
+ *                    merged.
+ *
+ * public void merge(ACC accumulator, java.lang.Iterable<ACC> iterable)
+ * }
+ * </pre>
+ *
+ * <pre>
+ * {@code
+ * Resets the accumulator for this TableAggregateFunction. This function must be implemented for
+ * data set grouping aggregates.
+ *
+ * param: accumulator the accumulator which needs to be reset
+ *
+ * public void resetAccumulator(ACC accumulator)
+ * }
+ * </pre>
+ *
+ * <pre>
+ * {@code
+ * Output data incrementally in upsert or append mode. For example, if we emit data for a TopN
+ * TableAggregateFunction, we don't need to output all top N elements each time a record comes.
+ * It is more efficient to output data incrementally in upsert mode, i.e, only output data whose
+ * rank has been changed.
+ *
+ * param: accumulator           the accumulator which contains the current aggregated results
+ * param: out                   the collector used to output data.
+ *
+ * public void emitValue(ACC accumulator, RetractableCollector out)
```

Comment:
I think `emitValue` just like `getValue` always get complete full results. And `emitValueWithRetract` should get the incrementally result. Please see the detail example in [FLIP-29](https://cwiki.apache.org/confluence/pages/viewpage.action?pageId=97552739)